service/common/src/main/java/org/apache/polaris/service/exception/IcebergPersistenceExceptionMapper.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.exception;
+
+import jakarta.persistence.PersistenceException;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import jakarta.ws.rs.ext.Provider;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+
+
+@Provider
+public class IcebergPersistenceExceptionMapper
+    implements ExceptionMapper<PersistenceException> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IcebergPersistenceExceptionMapper.class);
+
+  @Override
+  public Response toResponse(PersistenceException exception) {
+    LOGGER.info("Full PersistenceException", exception);
+
+    ErrorResponse icebergErrorResponse =
+        ErrorResponse.builder()
+            .responseCode(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode())
+            .withType(exception.getClass().getSimpleName())
+            .withMessage(exception.getMessage())
```

Comment:
The PR description mentions avoiding exposing internal database information in these errors, but will this message not contain database-specific information?

Do you have an example exception and corresponding JSON error response?

Reply:
This is absolutely right, my bad 🤦 
I can reproduce it in `PolarisEclipseLinkMetaStoreManagerTest.java` by messing with `persistence.xml`
<img width="1440" alt="Screenshot 2025-03-21 at 23 04 06" src="https://github.com/user-attachments/assets/654b8ac0-30ad-4376-a380-caa8bc36790a" />

Reply:
How can we be sure that `ExceptionUtils.getRootCause().getMessage()` (current code) does not expose internal database information?

Reply:
At this moment, I'd suggest to wrap the `jakarta.persistence.PersistenceException` in EclipseLink module in a RuntimeException, like what we did in JDBC impl. https://github.com/apache/polaris/blob/d3b24d2d1ce04d850f345ff947eb46093f663885/extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcBasePersistenceImpl.java#L98. 

In that case, we don't have to make any change to mappers.

Reply:
SGTM ^